tests/framework/components/mocks.py

```diff
@@ -92,20 +92,29 @@ def __init__(self, *args, name="test_listener"):
         self.collect_metrics_called = False
         self.simulation_end_called = False
 
+        self.time_step_prepare_index = None
+        self.time_step_index = None
+        self.time_step_cleanup_index = None
+        self.collect_metrics_index = None
```

Comment:
Should these like objects be contained in something, dict where keys are phases?

Reply:
yeah that is better